tests/homePageHeader.spec.js

```diff
@@ -0,0 +1,43 @@
+import { test, expect } from "@playwright/test";
+
+test ("Test redirect to what is new page", async({page}) => {
+    await page.goto("https://magento.softwaretestingboard.com/");
+    await page.locator("#ui-id-3").click();
+    const pageURL = page.url();
+    await expect(pageURL).toContain("/what-is-new");
+});
+
+test ("Test redirect to women page", async ({page}) => {
+    await page.goto("https://magento.softwaretestingboard.com/");
+    await page.locator("#ui-id-4").click();
+    const pageURL = page.url();
+    await expect(pageURL).toContain("/women");
+});
+
+test ("Test redirect to Men page", async ({page}) => {
+    await page.goto("https://magento.softwaretestingboard.com/");
+    await page.locator("#ui-id-5").click();
+    const pageURL = page.url();
+    await expect(pageURL).toContain("/men");
+});
+
+test ("Test redirect to Gear page", async({page}) => {
+    await page.goto("https://magento.softwaretestingboard.com/");
+    await page.locator("#ui-id-6");
+    const pageURL = page.url();
+    await expect(pageURL).toContain("/gear");
```

Comment:
Also , do not do this. This is a classic async versus sync mistake

```suggestion
    await page.locator("#ui-id-6").click()
    await expect(page.url()).toContain("/gear");
```

Your const pageURL line is a synchronous line and your await page locator is a asynchronous line
See my example below, I have two lines. First line I write with setTimeout is async and second line I write is with console.log which is sync. You can see below the answer, the order that I get the response.

![image](https://github.com/adrianmaciuc/break-it-playwright-javascript-project/assets/102796169/3c14b8cc-1177-4a77-9584-a9dff14c7f6b)

Notice that the second line is run first. Thats because sync just runs, where async has lower priority.

If you wait for a result such as what page url you should be after an async using the method .url(), then the method you should use also in an async way. A good way of doing this is to use the .url() method inside the expect, as suggested.

Reply:
ok, it makes sense why the tests have failed.